src/sentry/integrations/slack/threads/activity_notifications.py

```diff
@@ -15,3 +22,185 @@ class AssignedActivityNotification(BaseAssignedActivityNotification):
+                },
+            )
+
+            raise Exception(f"Activity type {activity_type} is incorrect")
+        self._activity: Activity = activity
+
+    def get_link(self) -> str:
+        """
+        Returns the link to where the issue was created in the external provider.
+        """
+        link = self._activity.data.get(self._URL_KEY, None)
+        if not link:
+            metrics.incr(
+                self._NO_LINK_KEY_METRICS,
+                sample_rate=1.0,
+            )
+
+            _default_logger.info(
+                "Activity does not have a url key, using fallback",
+                extra={
+                    "activity_id": self._activity.id,
+                },
+            )
+            link = ""
+        return link
+
+    def get_provider(self) -> str:
+        """
+        Returns the provider of the activity for where the issue was created.
+        Returns the value in lowercase to provider consistent value.
+        If key is not found, or value is empty, uses the fallback value.
+        """
+        provider = self._activity.data.get(self._PROVIDER_KEY, None)
+        if not provider:
+            metrics.incr(
+                self._NO_PROVIDER_KEY_METRICS,
+                sample_rate=1.0,
+            )
+            _default_logger.info(
+                "Activity does not have a provider key, using fallback",
+                extra={
+                    "activity_id": self._activity.id,
+                },
+            )
+            provider = self.DEFAULT_PROVIDER_FALLBACK_TEXT
+
+        return provider.lower()
+
+    def get_ticket_number(self) -> str:
+        """
+        Returns the ticket number for the issue that was created on the external provider.
+        """
+        ticket_number = self._activity.data.get(self._TICKET_KEY, None)
+        if not ticket_number:
+            metrics.incr(
+                self._NO_LABEL_KEY_METRICS,
+                sample_rate=1.0,
+            )
+            _default_logger.info(
+                "Activity does not have a label key, using fallback",
+                extra={
+                    "activity_id": self._activity.id,
+                },
+            )
+            ticket_number = ""
+
+        return ticket_number
+
+
+class _AsanaExternalIssueCreatedActivity(_ExternalIssueCreatedActivity):
+    """
+    Override class for Asana as, at this time, the label, or ticket number, does not exist and has to be derived.
+    If plausible, this could be removed if the activity object itself properly has the correct data, but side effects
+    for that change are not yet known.
+    """
+
+    _DEFAULT_ASANA_LABEL_VALUE = "Asana Issue"
+
+    def get_ticket_number(self) -> str:
+        # Try to use the base logic if it works as a just in-case
+        stored_value = super().get_ticket_number()
+        if stored_value != "" and stored_value != self._DEFAULT_ASANA_LABEL_VALUE:
+            return stored_value
```

Comment:
should we be tracking how often this happens?

Reply:
It would probably just be easier to do a data check through ReDash and check the values. Tracking/metrics on this wouldn't help us prevent anything anyways or have us actively take action